include/dd/Package.hpp

```diff
@@ -485,6 +485,69 @@ template <class Config> class Package {
     return f;
   }
 
+  // generate general GHZ state with n qubits
+  vEdge makeGHZState(const std::size_t n) {
+    if (n > nqubits) {
+      throw std::runtime_error{
+          "Requested state with " + std::to_string(n) +
+          " qubits, but current package configuration only supports up to " +
+          std::to_string(nqubits) +
+          " qubits. Please allocate a larger package instance."};
+    }
+
+    if (n == 0U) {
+      return vEdge::one();
+    }
+
+    auto leftSubtree = vEdge::one();
+    auto rightSubtree = vEdge::one();
+
+    for (std::size_t p = 0; p < n - 1; ++p) {
+      leftSubtree = makeDDNode(static_cast<Qubit>(p),
+                               std::array{leftSubtree, vEdge::zero()});
+      rightSubtree = makeDDNode(static_cast<Qubit>(p),
+                                std::array{vEdge::zero(), rightSubtree});
+    }
+
+    return makeDDNode(
+        static_cast<Qubit>(n - 1),
+        std::array<vEdge, RADIX>{
+            {{leftSubtree.p, {&constants::sqrt2over2, &constants::zero}},
+             {rightSubtree.p, {&constants::sqrt2over2, &constants::zero}}}});
+  }
+
+  // generate general W state with n qubits
+  vEdge makeWState(const std::size_t n) {
+    if (n > nqubits) {
+      throw std::runtime_error{
+          "Requested state with " + std::to_string(n) +
+          " qubits, but current package configuration only supports up to " +
+          std::to_string(nqubits) +
+          " qubits. Please allocate a larger package instance."};
+    }
+
+    if (n == 0U) {
+      return vEdge::one();
+    }
+
+    auto leftSubtree = vEdge::zero();
+    // This is just a temporary trick to get the scaling right.
+    // It is inherently limited by the tolerance we employ and will crash once
+    // 1/sqrt(n) < RealNumber::eps.
+    // The trick to resolve this is to spread the 1/sqrt(n) across the various
+    // levels of the DD during construction.
```

Comment:
Based on the discussion we had last time, this comment is outdated and can be removed.
Instead of that, IIRC, we discussed to add a check that warns whenever `1./sqrt(n) < RealNumber::eps` because that's when the method falls apart. A potential warning message for the exception too throw could include something like "Requested qubit size for generating W-state would lead to an underflow due to `1 / sqrt(n)` being smaller than the currently set tolerance (<actual tolerance>). If you still wanna run the computation, please lower the tolerance accordingly".

Reply:
Done ✅